src/src/shared/definitions.rs

```diff
@@ -0,0 +1,65 @@
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct TableId(pub String);
+
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct PageId(pub usize);
+
+pub mod page {
```

Comment:
Instead of declaring page under definitions, I think page deserves its own file (page.rs)